src/components/UserProfile/AddNewUserProfile/UserProfileAdd.jsx

```diff
@@ -573,17 +573,15 @@ class AddUserProfile extends Component {
           .then(res => {
             if (res.data.warning) {
               toast.warn(res.data.warning);
-            } else {
-              if (
-                this.checkIfDuplicate(userData.firstName, userData.lastName) &&
-                !allowsDuplicateName
-              ) {
-                this.setState({
-                  popupOpen: true,
-                });
-                return;
-              }
+            } else if (
+              this.checkIfDuplicate(userData.firstName, userData.lastName) &&
+              !allowsDuplicateName
+            ) {
+              this.setState({
+                popupOpen: true,
+              });
               toast.success('User profile created.');
+              return;
```

Comment:
Also, I find some issues in this duplicate name flow. Please go through the video to understand it.

https://www.dropbox.com/s/d3u73ls3yya0jec/Alan_PR_review.mov?dl=0

@LeeSingChanYau  Could you please test your code for all these flows before submitting for next review ? Thanks and it will reduce the time spent on repeated reviews. 

@ Everyone: These are the major testcases/usecases for this PR.
case 1. Create new user case 2. Create new user with duplicate name case a. Confirm case b.Change name 3. Filter and create user following case 1 and case 2 a, case 2 b. For all these cases, see if the user list gets updated accordingly.

Reply:
@RJBornToWin thank you for the feedback. Please review. Changed conditional statements when calling createUser. Now we return when this.checkIfDuplicate(userData.firstName, userData.lastName) && !allowsDuplicateName) is met, because a the popup will open and we dont to create the user just yet. If we allow duplicateName and
 there are no warnings then we will call toast.success().
 
 If you have a better idea on how to implement the conditions, please leave code suggestions if possible.